app/assets/javascripts/application/donor_form.js

```diff
@@ -50,4 +46,49 @@ $(function() {
         format:'H:i'
     });
 
+    function checkEmailExists(email) {
+        return $.ajax({
+            url: '/users/exists',
+            dataType: 'json',
+            type: 'GET',
+            data: {email: email},
+            success: function(data) {
+                // console.log('Query success!');
+            }.bind(this),
+            error: function(xhr, status, err) {
+                console.error(window.location.href, status, err.toString());
+            }.bind(this)
+        });
+    }
+
+
+    $('#donor-form-fields')
+        .on('invalid.fndtn.abide', function () {
+            toastr.error('There was an error with your submission');
+        })
+        .on('valid.fndtn.abide', function () {
+            if (gon.isSignedInOnLoad) {
+                // When the user makes a GET request on this page, the user is already signed in
+                donationModalInstance.handleDonation(function(xhr) {
+                    xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content'))
+                });
+            } else {
+                var dataArray = $('#donor-form-fields').serializeArray();
+                var email = _.find(dataArray, function(field) {return field.name === 'donation[email]'}).value;
+
+                checkEmailExists(email)
+                .done(function(result) {
+                    // console.log(result);
+                    // If 'result' is null, no email was found so the user must register
+                    // If 'result' is not null, then 'result' is an object {email: <email>}
+                    if (donationModalInstance.isMounted()) {
+                         _.isNull(result) ? donationModalInstance.setState({email: email, userExists: false}) : donationModalInstance.setState({email: email, userExists: true})
+                    }
+                    donationModalInstance.handleShowModal();
+                })
+                .fail(function(result) {
+                    toastr.error("We're sorry, something went wrong. Please try again.");
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>'toastr' is not defined.